contracts/FlashRolloverV1toV2.sol

```diff
@@ -0,0 +1,364 @@
+                bundleERC1155Holdings[i] = ERC1155Holding(tokenAddr, tokenId, amount);
+            } catch { break; }
+        }
+
+        sourceAssetWrapper.withdraw(oldBundleId);
+
+        // Create new asset vault
+        address vault = address(uint160(vaultId));
+
+        for (uint256 i = 0; i < bundleERC721Holdings.length; i++) {
+            ERC721Holding memory h = bundleERC721Holdings[i];
+
+            if (h.tokenAddress == address(0)) {
+                break;
+            }
+
+            IERC721(h.tokenAddress).safeTransferFrom(address(this), vault, h.tokenId);
+        }
+
+        for (uint256 i = 0; i < bundleERC1155Holdings.length; i++) {
+            ERC1155Holding memory h = bundleERC1155Holdings[i];
+
+            if (h.tokenAddress == address(0)) {
+                break;
+            }
+
+            IERC1155(h.tokenAddress).safeTransferFrom(address(this), vault, h.tokenId, h.amount, bytes(""));
+        }
+    }
+
+    function _getContracts(RolloverContractParams memory contracts) internal returns (OperationContracts memory) {
+        return
+            OperationContracts({
+                loanCore: contracts.sourceLoanCore,
+                borrowerNote: contracts.sourceLoanCore.borrowerNote(),
+                lenderNote: contracts.sourceLoanCore.lenderNote(),
+                feeController: contracts.targetLoanCore.feeController(),
+                sourceAssetWrapper: contracts.sourceLoanCore.collateralToken(),
+                targetVaultFactory: contracts.targetVaultFactory,
+                repaymentController: contracts.sourceRepaymentController,
+                originationController: contracts.targetOriginationController,
+                targetLoanCore: contracts.targetLoanCore,
+                targetBorrowerNote: contracts.targetLoanCore.borrowerNote()
+            });
+    }
+
+    function _validateRollover(
+        ILoanCoreV1 sourceLoanCore,
+        IVaultFactory targetVaultFactory,
+        LoanLibraryV1.LoanTerms memory sourceLoanTerms,
+        LoanLibrary.LoanTerms calldata newLoanTerms,
+        uint256 borrowerNoteId
+    ) internal {
+        require(sourceLoanCore.borrowerNote().ownerOf(borrowerNoteId) == msg.sender, "caller not borrower");
+        require(newLoanTerms.payableCurrency == sourceLoanTerms.payableCurrency, "currency mismatch");
+        require(newLoanTerms.collateralAddress == address(targetVaultFactory), "must use vault");
+    }
+
+    function setOwner(address _owner) external override {
+        require(msg.sender == owner, "not owner");
+
+        owner = _owner;
+
+        emit SetOwner(owner);
+    }
+
+    function flushToken(IERC20 token, address to) external override {
+        require(msg.sender == owner, "not owner");
+
+        uint256 balance = token.balanceOf(address(this));
+        require(balance > 0, "no balance");
+
+        token.transfer(to, balance);
+    }
+
+    receive() external payable {}
```

Comment:
This should actually never need to be used (since we're not withdrawing eth into a bundle). Maybe cleaner if I just delete the fallback function then?